tensordict/base.py

```diff
@@ -4709,8 +4766,12 @@ def _values_list(
                 is_leaf=is_leaf,
                 collapse=collapse,
             )
-            source = dict(zip(keys, vals))
-            return [source[key] for key in sorting_keys]
+            if torch.compiler.is_dynamo_compiling():
+                key_to_index = {key: i for i, key in enumerate(keys)}
```

Comment:
Interesting so the `dict` construct does not work when compiling?

Reply:
@anijain2305 may have more context but my understanding is just that this isn't registered. There's an item in the TD tracker for this!